numpy/lib/format.py

```diff
@@ -604,7 +604,6 @@ def _read_array_header(fp, version):
         raise ValueError(msg.format(d))
 
     if EXPECTED_KEYS != d.keys():
-        keys = sorted(d.keys())
         msg = "Header does not contain the correct keys: {!r}"
         raise ValueError(msg.format(d.keys()))
```

Comment:
```suggestion
        keys = sorted(d.keys())
        msg = "Header does not contain the correct keys: {!r}"
        raise ValueError(msg.format(keys))
```
I suspect this might have been the original intention.

Reply:
hmm, yes that does make sense.
Do you want me to commit the suggestion ?